### PR TITLE
feat: add support for Github Enterprise

### PR DESCRIPTION
**Pull request by feliperoveran**
Workflow runs can be found here: https://github.com/feliperoveran/action-semantic-pull-request/actions

Works like a charm on GHES by providing:
```
with:
  githubServerUrl: https://github.myorg.com/api/v3
```

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -5,17 +5,23 @@ const validatePrTitle = require('./validatePrTitle');
...
-      validateSingleCommit
+      validateSingleCommit,
+      githubServerUrl
     } = parseConfig();
 
+    const client = githubServerUrl
+      ? github.getOctokit(process.env.GITHUB_TOKEN)
+      : github.getOctokit(process.env.GITHUB_TOKEN, {
+        baseUrl: githubServerUrl
+      });
```
Comment (amannn) [inline src/index.js:23]: Do you by chance know if the octokit client uses the default when `baseUrl` is `undefined`?

Maybe we could simplify it to just:

```suggestion
    const client = github.getOctokit(process.env.GITHUB_TOKEN, {
      baseUrl: githubServerUrl
    });
```

File: action.yml [hunk 2]
```diff
@@ -29,3 +29,6 @@ inputs:
   validateSingleCommit:
     description: "When using \"Squash and merge\" on a PR with only one commit, GitHub will suggest using that commit message instead of the PR title for the merge commit, and it's easy to commit this by mistake. Enable this option to also validate the commit message for one commit PRs." 
     required: false
+  githubServerUrl:
+    description: "Github Enterprise server URL. Set this if using GHES. E.g. https://github.myorg.com/api/v3"
```
Comment (amannn) [inline action.yml:33]: What do you think about using `githubBaseUrl` as the name to stick to the naming convention from the octokit client?

```suggestion
  githubBaseUrl:
    description: "If you use Github Enterprise, you can set this to the URL of your server (e.g. https://github.myorg.com/api/v3)"
```

Can you please also document this new parameter in the README?

**Review by amannn — COMMENTED** [review]
Thanks a lot for this excellent PR! I only left two minor comments.

File: src/index.js (hunk 1 above)
Comment (feliperoveran) [reply src/index.js]: I wasn't sure but I gave it a try and it does indeed uses the default :)

File: action.yml (hunk 2 above)
Comment (feliperoveran) [reply action.yml]: renamed to `githubBaseUrl`. thanks for the suggestion!

also added it to the README

**Review by amannn — APPROVED** [review]
Looks really good, thank you so much for your contribution! 👏

**Pull request merged**